src/loaders/abstract-loader.coffee

```diff
@@ -276,23 +277,26 @@ class AbstractLoader
     promises = []
 
     for association in @additionalIncludes
-      loadOptions =
-        cache: @loadOptions.cache
-        headers: @loadOptions.headers
-        only: association.ids
-        params:
-          apply_default_filters: false
-
-      if association.collection
-        promises.push association.collection.fetch(loadOptions)
-      else
-        collectionName = @_getModel().associationDetails(association.name).collectionName
-        if association.loadOptions
-          loadOptions = _.extend(loadOptions, association.loadOptions)
+      batches = Utils.chunk(association.ids, @associationIdLimit)
+
+      batches.forEach (batchedIds) =>
```

Comment:
FYI: We should consider breaking up this method. We're now in a double nested loop.

Reply:
Good idea